sdk/go/src/sawtooth_sdk/client/encoder.go

```diff
@@ -0,0 +1,156 @@
+package client
+
+import (
+	"encoding/hex"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	"sawtooth_sdk/protobuf/batch_pb2"
+	"sawtooth_sdk/protobuf/transaction_pb2"
+	"strings"
+	"time"
+)
+
+// See sdk/go/tests/client_test.go for example usage.
+
+type Transaction struct{ pb *transaction_pb2.Transaction }
+type Batch struct{ pb *batch_pb2.Batch }
+
+type TransactionEncoder struct {
+	privkey         []byte
+	pubkey          []byte
+	familyName      string
+	familyVersion   string
+	payloadEncoding string
+	addNonce        bool
+}
```

Comment:
In the JS Client SDK, no decisions are made about what defaults are stored in the TransactionEncoder. So for example, even though `inputs` and `outputs` are likely to be different in each transaction, you have the option to set defaults if you like. Some possible use cases would be if wild cards are used to designate a whole subtree, or a Transaction Processor stores all of one user's data at one address.

Which is a long way of saying, I think you should be able to set defaults for `inputs` and `outputs`. There is more of a case for omitting `dependencies`, but at that point it is the odd-one-out, and I think just for consistency of the interface, it should be possible to set that too.